### PR TITLE
fix: update max slippage state

### PR DESCRIPTION
**Pull request by zzmp**
- Makes max slippage resettable
- Stores max slippage as a number, and exposes it through new useMaxSlippage hook (which includes auto)

### CODE REVIEW

File: src/lib/components/Swap/Settings/MaxSlippageSelect.tsx [hunk 1]
```diff
@@ -98,61 +98,69 @@ const Warning = memo(function Warning({ state, showTooltip }: { state: WarningSt
...
 export default function MaxSlippageSelect() {
-  const [maxSlippage, setMaxSlippage] = useAtom(maxSlippageAtom)
-
-  const [custom, setCustom] = useState('')
   const input = useRef<HTMLInputElement>(null)
   const focus = useCallback(() => input.current?.focus(), [input])
 
+  const [autoSlippage, setAutoSlippage] = useAtom(autoSlippageAtom)
+  const [maxSlippage, setMaxSlippage] = useAtom(maxSlippageAtom)
+  const value = useMemo(() => maxSlippage?.toString() || '', [maxSlippage])
```
Comment (JFrankfurt) [inline src/lib/components/Swap/Settings/MaxSlippageSelect.tsx:106]: nit: use a more descriptive variable name. Maybe `maxSlippageString`?

File: src/lib/components/Swap/Settings/MaxSlippageSelect.tsx [hunk 2]
```diff
@@ -98,61 +98,69 @@ const Warning = memo(function Warning({ state, showTooltip }: { state: WarningSt
...
           {...tooltipProps}
         >
           <Row color={warning === WarningState.INVALID_SLIPPAGE ? 'error' : undefined}>
             <DecimalInput
-              size={Math.max(custom.length, 3)}
-              value={custom}
-              onChange={setCustom}
+              size={Math.max(value.length, 3)}
+              value={value}
+              onChange={(input) => processInput(+input)}
```
Comment (JFrankfurt) [inline src/lib/components/Swap/Settings/MaxSlippageSelect.tsx:163]: nit - kinda feels like the typing logic should be happening in the processInput function

**Review by ianlapham — APPROVED** [review]

**Review by JFrankfurt — APPROVED** [review]

File: src/lib/components/Swap/Settings/MaxSlippageSelect.tsx (hunk 2 above)
Comment (zzmp) [reply src/lib/components/Swap/Settings/MaxSlippageSelect.tsx:163]: `DecimalInput` takes a `string`, but I still want to store `maxSlippage` as a number so it is better typed. Making `processInput` take a `number` lets it be reused for `onInputSelect`/mounting logic as well.

File: src/lib/components/Swap/Settings/MaxSlippageSelect.tsx (hunk 1 above)
Comment (zzmp) [reply src/lib/components/Swap/Settings/MaxSlippageSelect.tsx]: Used `maxSlippageInput`

**Pull request merged**